src/tools/ogmaTool.ts

```diff
@@ -98,22 +108,35 @@ export class OgmaTools {
     return items.isNode;
   }
 
-  public static topologicalSort(nodes: NodeList): NodeId[] {
+  /**
+   * Transform a nodelist to an array of node ids sorted by position in a chain and the length of the chain
+   * A chain is: (a)-(b)-(c)-(d)
+   */
+  public static topologicalSort(nodes: NodeList): {chain: NodeId[]; numberOfChain: number} {
```

Comment:
if a chain is always linear why not just do getFirstNode and then the while loop (and remove the foreach).

if a chain can be like this (a)-(b)-(c)-(d) but also b and c are linked to e. Then with this logic it feels like e could be forgotten. a and d would be startOfChains. You'd start with a then b then c then d. d has no adjacent nodes not visited so you stop the while loop. In the foreach, you start again form d (again, feels you can remove the foreach since d has already been visited), since d has been visited, you exit the while loop and the foreach. But e has been forgotten. Let's discuss haha, pretty sure my comment is super unclear

Reply:
> if a chain can be like this (a)-(b)-(c)-(d) but also b and c are linked to e.

if b or c are linked to another node (e) it is no longer a chain and this method should not be used.
 
 Also nodes are possibly multiple chains  (a)-(b)-(c) and (d)-(e) ...

Reply:
It is not clear fo me the part about removing the foreach

Reply:
from you answer above, it's clearer and we cannot remove the foreach